argocd-cli.yaml

```diff
@@ -0,0 +1,46 @@
+package:
+  name: argocd-cli
+  version: 2.6.2
+  epoch: 0
+  description: "Declarative continuous deployment for Kubernetes."
+  target-architecture:
+    - all
+  copyright:
+    - paths:
+        - "*"
+      license: Apache License 2.0
+  dependencies:
+    runtime:
+      - ca-certificates-bundle
+environment:
+  contents:
+    packages:
+      - ca-certificates-bundle
+      - busybox
+      - go
+      - jq
+pipeline:
+  - uses: git-checkout
+    with:
+      repository: https://github.com/argoproj/argo-cd
+      branch: v${{package.version}}
+      destination: argo-cd
+  #TODO(kamesh) - simplify this using go/build pipeline
+  # but it requires few OOTB variables
+  - name: build argocd cli
+    runs: |
+      set +x
+      cd argo-cd
+      DEST_DIR="${{targets.destdir}}/usr/bin"
+      mkdir -p "$DEST_DIR"
+      build_date=$(date -u +'%Y-%m-%dT%H:%M:%SZ')
+      git_commit=$(git rev-parse HEAD)
+      git_tree_state=$(git status --porcelain)
+      kubectl_version=$(go list -m k8s.io/client-go | head -n 1 | rev | cut -d' ' -f1 | rev)
+      CGO_ENABLED=0 go build -v \
+        -trimpath -ldflags \
+        "-X github.com/argoproj/argo-cd/v2/common.version=${{package.version}} -X github.com/argoproj/argo-cd/v2/common.buildDate==${build_date} -X github.com/argoproj/argo-cd/v2/common.gitCommit=${git_commit} -X github.com/argoproj/argo-cd/v2/common.gitTreeState=${git_tree_state} -X github.com/argoproj/argo-cd/v2/common.kubectlVersion=${kubectl_version}" \
+        -o "${{targets.destdir}}/usr/bin/argocd" ./cmd
+  - name: ensure binary has right version info
+    runs: "set -eux \nARGOCD_CLI=${{targets.destdir}}/usr/bin/argocd\nkubectl_version=$(go list -m k8s.io/client-go | head -n 1 | rev | cut -d' ' -f1 | rev)\n_argocd_version=$($ARGOCD_CLI version --client -o json | jq -r '.client.Version')\n_git_commit=$($ARGOCD_CLI version --client -o json | jq -r '.client.GitCommit')\n_kubectl_version=$($ARGOCD_CLI version --client -o json | jq -r '.client.KubectlVersion')\n# the git tree dirty as the make generates dist dir and dist is not gitignored\n[ \"$_argocd_version\" = \"v${{package.version}}+$(echo \"${_git_commit}\"|cut -c1-7).dirty\" ] || echo \"versions not matching\"\n[ \"$_kubectl_version\" = \"$kubectl_version\" ] || echo \"kubectl versions not matching\"\n"
```

Comment:
i agree there should be test but not sure we should do it here. also this is a bit too long & hard to read

Reply:
simplified it to test only built version